.github/workflows/docs.yml

```diff
@@ -5,15 +5,21 @@ name: Documentation
 
 on: [push, pull_request]
 
+permissions:
+  contents: read
+
 env:
   DOXYGEN_VERSION: 1.9.6
 
 jobs:
   build:
+    name: Build documentation
```

Comment:
`documentation` is redundant here since it is under a workflow named `Documentation` and displayed as `Documentation / Build documentation`.

Also, this change should be in a separate commit so that we can cleanly revert the permission related changes if there are any problems with it.

Reply:
I have dropped the `documentation` part from the name and moved to a separate commit.
The names don't bring much value, so I could drop them completely if preferred.